src/main/java/org/mule/extensions/jms/api/connection/factory/jndi/JndiConnectionFactory.java

```diff
@@ -85,25 +87,22 @@
     }
   }
 
-
   @Override
-  public void initialise() throws InitialisationException {
-
+  public ConnectionFactory getTargetConnectionFactory() {
     try {
-      setupNameResolver();
-
       Object temp = getJndiNameResolver().lookup(connectionFactoryJndiName);
       if (temp instanceof ConnectionFactory) {
-        this.setTargetConnectionFactory((ConnectionFactory) temp);
-      } else {
-        throw new IllegalArgumentException("No valid ConnectionFactory was provided. Unable to initialise.");
+        return (ConnectionFactory) temp;
```

Comment:
I think this was done only once before. Maybe it should be cached instead of doing a lookup every time this is called?